### ISSUE TITLE
Possibility to specify a conversion of the output when passing a custom objective function

### ISSUE THREAD

**Issue by mathiasottenbreit** [microsoft/LightGBM#6292]
Some loss functions require a conversion of the output, such as in your built-in poisson, gamma or tweedie objectives (the ConvertOutput() method in your source code). Currently, custom objective functions return only the gradient and hessian so that it is not possible to specify the corresponding conversion of the output. Can you add this please? Thanks.
